SU-HSDO/suhumsci
# SHS-5907: Implementation: Editors can easily set "view" access for Private Pages

# READY FOR REVIEW

## Summary
- Adds a new contrib module `content_access_simple` that adds a simpler interface for settings view permissions per node.

## Need Review By (Date)
2024-11-06

## Urgency
low

## Steps to Test
1. Login to the site as an admin
2. Create a Private Page
3. Title it
4. Verify there is a new area below all of the fields called "Access and Permissions"

![Screenshot 2024-11-05 at 2 01 24 PM](https://github.com/user-attachments/assets/56defff3-e5ea-4069-b378-cc24d5652745)

6. Open it up and verify that there is a list of roles under "Visibility" but there is not "Developer", "Anonymous", or "Authenticated".
7. Choose a role, like "Site Manager" and enable view access
8. Save the node
9. Visit the "Access control" tab for this node
10. Verify that under "View any hs_private_page content", the "Site Manager" is checked in addition to "Developer"  and "Authenticated user" (which were the defaults originally but they are hidden as per config)
11. Uncheck "Authenticated user" under view any here, and save
12. Attempt to view the page on a logged in user who is not "Site Manager" role and verify that you cannot access it
13. Attempt to view the page on a logged in user who is "Site Manager" and verify that you can access it
14. Edit the node and verify that under "Access and Permissions" you get a "complex" warning and no roles are listed

![Screenshot 2024-11-05 at 1 59 39 PM](https://github.com/user-attachments/assets/09fe4cfa-15fa-4472-b653-5e7d3684a395)

16. To fix, visit the content type permissions at `/admin/structure/types/manage/hs_private_page/access` and also uncheck "Authenticated user" and save
17. Edit the node again and this time, and this time, mark the node as unpublished and save
18. Edit the node and verify that under "Access and Permissions" you get a "unpublished" warning but the roles are still listed

![Screenshot 2024-11-05 at 1 59 19 PM](https://github.com/user-attachments/assets/49d6b5a3-a429-4239-916b-4c0b8497066d)

20. Visit `/admin/structure/types/manage/hs_private_page/form-display` and drag the "Content Access Simple" "field" to another place in the form and save
21. Edit the node and verify that the Access and Permissions has moved.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208890594589119
